### PR TITLE
[Api] Add basic APIs

### PR DESCRIPTION
**Pull request by hereisnaman**
Added first set of APIs for:

- getMe
- getPlayer
- getPlayerFollowers (paginated)
- getPlayerFollowings (paginated)
- followPlayer
- unfollowPlayer

### CODE REVIEW

File: apps/api/controllers/players/:playerId.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import BaseController, { NotFoundError } from 'api/controllers/base/controller';
+import { Player } from 'models/player';
+
+export default class PlayerController extends BaseController {
+  Get = async () => {
+    const player = await Player.findOne(Number(this.req.params.playerId));
+    if (!player) throw new NotFoundError('Player');
+
+    this.res.send({ data: { player } });
```
Comment (jatinkatyal13) [inline apps/api/controllers/players/:playerId.ts:9]: this.res.json

Comment (hereisnaman) [reply apps/api/controllers/players/:playerId.ts:9]: headers are automatically set to applications/json in case of objects.

Comment (jatinkatyal13) [reply apps/api/controllers/players/:playerId.ts:9]: still it'll just be more verbose

File: apps/api/controllers/base/controller.ts [hunk 2]
```diff
@@ -1,20 +1,89 @@
...
+
+  static register(router) {
+    // NOTE(naman) hacked to be used only on child classes
+    const Controller = this as any;
+    const instance = new Controller() as BaseController;
+
+    const methods = ['Get', 'Post', 'Put', 'Patch', 'Delete'];
+
+    methods.forEach((method) => {
+      if (instance.hasOwnProperty(method)) {
```
Comment (jatinkatyal13) [inline apps/api/controllers/base/controller.ts:38]: Let's have Method not Supported as the default handler?

File: apps/api/controllers/base/controller.ts [hunk 3]
```diff
@@ -1,20 +1,89 @@
...
+    return this.processStack(stack, next);
+  };
+
+  private processStack = async (stack: Middleware[], next: () => any) => {
+    const nextWrapper = async (index) => {
+      index++;
+
+      if (index === stack.length) return next();
+
+      return await stack[index](this.req, this.res, () => nextWrapper(index));
```
Comment (jatinkatyal13) [inline apps/api/controllers/base/controller.ts:62]: return await kyu?

File: apps/api/index.ts [hunk 4]
```diff
@@ -19,14 +20,12 @@ import errorHandler from 'api/middlewares/errorHandler';
   app.use(bodyParser.urlencoded({ extended: true }));
   app.use(compression());
 
-  app.use('/api/', await routes);
+  app.use('/api/', authenticate, await routes);
```
Comment (jatinkatyal13) [inline apps/api/index.ts:23]: open endpoints k lie can't we have a `authenticateOrPass` middleware?

while processing a request, only the root file will be using the passport middleware, rest of the handlers will use `req?.user` for authorization

File: domains/players/index.ts [hunk 5]
```diff
@@ -0,0 +1,7 @@
+import { PlayersFollowAssociation } from './followAssociation';
+
+export class Players {
```
Comment (jatinkatyal13) [inline domains/players/index.ts:3]: why do we need class here?

File: apps/api/controllers/base/controller.ts (hunk 2 above)
Comment (hereisnaman) [reply apps/api/controllers/base/controller.ts:38]: Yeah, we can add that in future.

File: apps/api/controllers/base/controller.ts (hunk 3 above)
Comment (hereisnaman) [reply apps/api/controllers/base/controller.ts:62]: error boundary daalni he ek alg se baad mein.

File: apps/api/index.ts (hunk 4 above)
Comment (hereisnaman) [reply apps/api/index.ts:23]: we will worry about that when we will introduce auth.

File: domains/players/index.ts (hunk 5 above)
Comment (hereisnaman) [reply domains/players/index.ts:3]: all domains will expose a facade class. That is what implements a bounded context in ddd.

**Review by jatinkatyal13 — APPROVED** [review]

**Pull request merged**